extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -180,8 +188,15 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
 
-                return getActiveConnections(a).size()
-                     - getActiveConnections(b).size();
+                // Get connection weight for the two systems being compared.                
+                int weightA = a.getConnectionWeight().intValue();
+                int weightB = b.getConnectionWeight().intValue();
+
+                // Get current active connections, add 1 to both to avoid calculations with 0.
```

Comment:
Nope - sorting connections with zero active connections by their weight makes sense, but if that's what you're aiming for here, then perhaps that's what the comment should describe?

The thing that wouldn't make sense is arbitrarily avoiding zero.